include/usbg/usbg.h

```diff
@@ -54,6 +54,10 @@ extern "C" {
 #define USBG_MAX_NAME_LENGTH 40
 /* Dev name for ffs is a part of function name, we subtract 4 char for "ffs." */
 #define USBG_MAX_DEV_LENGTH (USBG_MAX_NAME_LENGTH - 4)
+/* ConfigFS just like SysFS uses page size as max size of file content */
+#define USBG_MAX_FILE_SIZE 4096
+/* OS Descriptors OS String identifier length */
+#define USBG_OS_STRING_QW_SIGN_LEN 14
```

Comment:
Please correct me if I'm wrong but shouldn't this be 15  because of terminating \0?
usbg_read_string_limited() is going to truncate the string to fit it inside the array so if the qw_sign is exactly 14 bytes it will be truncated to fit also trailing \0

Reply:
Hm, the kernel uses 14 internally too (include/linux/usb/composite.h). The kernel uses memcpy etc, so does not store it null terminated. But it stores it in UTF-16, whereas the kernel sysfs and libusbgx use UTF-8...
According to spec the signature should be the 7 character string "MSFT100" and I guess the only thing which might change is the version (100). So technically we could just allocate 8 bytes and we should be good. What do you think?

Reply:
You are right. I forgot about UTF-8 to UTF-16 conversion

Reply:
FWIW, in v2 I got rid of the static allocation of qw_sign anyway.